android/engine/src/main/java/org/smartregister/fhircore/engine/ui/login/LoginActivity.kt

```diff
@@ -42,9 +44,18 @@ class LoginActivity :
 
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
-    loginService.loginActivity = this
+    loginService.runningActivity = this
     loginViewModel.apply {
-      navigateToHome.observe(this@LoginActivity, { loginService.navigateToHome() })
+      navigateToHome.observe(
```

Comment:
What if a user already did the set-up pin process once?. There should be check here to check if user saved the pin previously or not.